internal/tasks/repo_update.go

```diff
@@ -0,0 +1,127 @@
+package tasks
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/flightctl/flightctl/api/v1alpha1"
+	api "github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/store"
+	"github.com/flightctl/flightctl/internal/store/model"
+	"github.com/flightctl/flightctl/pkg/log"
+	"github.com/flightctl/flightctl/pkg/reqid"
+	"github.com/go-chi/chi/v5/middleware"
+	"github.com/sirupsen/logrus"
+)
+
+func RepositoryUpdate(taskManager TaskManager) {
+	for {
+		select {
+		case <-taskManager.ctx.Done():
+			taskManager.log.Info("Received ctx.Done(), stopping")
+			return
+		case resourceRef := <-taskManager.channels[ChannelRepositoryUpdates]:
+			requestID := reqid.NextRequestID()
+			ctx := context.WithValue(context.Background(), middleware.RequestIDKey, requestID)
+			log := log.WithReqIDFromCtx(ctx, taskManager.log)
+			logic := NewRepositoryUpdateLogic(taskManager, log, taskManager.store, resourceRef)
+
+			switch {
+			case resourceRef.Op == RepositoryUpdateOpUpdate && resourceRef.Kind == model.RepositoryKind:
+				logic.HandleRepositoryUpdate(ctx)
+			case resourceRef.Op == RepositoryUpdateOpDeleteAll && resourceRef.Kind == model.RepositoryKind:
+				logic.HandleAllRepositoriesDeleted(ctx)
+			default:
+				log.Errorf("RepositoryUpdate called with unexpected kind %s and op %s", resourceRef.Kind, resourceRef.Op)
+			}
+		}
+	}
+}
+
+type RepositoryUpdateLogic struct {
+	taskManager TaskManager
+	log         logrus.FieldLogger
+	store       store.Store
+	resourceRef ResourceReference
+}
+
+func NewRepositoryUpdateLogic(taskManager TaskManager, log logrus.FieldLogger, store store.Store, resourceRef ResourceReference) RepositoryUpdateLogic {
+	return RepositoryUpdateLogic{taskManager: taskManager, log: log, store: store, resourceRef: resourceRef}
+}
+
+func (t *RepositoryUpdateLogic) HandleRepositoryUpdate(ctx context.Context) {
```

Comment:
why not return errors here then allow caller to decide log or handle?

Reply:
Done